### PR TITLE
[HttpFoundation] Fix session-related BC break

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 2.7
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #24941, #24934, #24947 and #24946
| License       | MIT
| Doc PR        | -

Conservative fix.

### CODE REVIEW

**Comment by sroze** [comment]
See #24947 and #24946. They've been given a 👍  in issues on top of that.

**Comment by nicolas-grekas** [comment]
@sroze yes but  prefer this approach, as it's more conservative: the diff with pre-7.2-concerns is much lighter, thus easier to validate.

File: src/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php [hunk 1]
```diff
@@ -340,7 +335,7 @@ public function isStarted()
      */
     public function setOptions(array $options)
     {
-        if (headers_sent()) {
+        if (headers_sent() || (\PHP_VERSION_ID >= 50400 && \PHP_SESSION_ACTIVE === session_status())) {
```
Comment (sroze) [inline src/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php]: Could you move that in a private method?

**Review by sroze — COMMENTED** [review]
Also, it would be nice to add a test for the fixed issue with the null session handler. You've got one you can copy/paste from my other PR.

**Comment by sroze** [comment]
@nicolas-grekas I don't mind either tbh so... 👍

**Comment by nicolas-grekas** [comment]
I borrowed you test case.
No need for the private method, that's diff boilerplate to me, and will be reduced to one instruction on master.

**Comment by sroze** [comment]
> will be reduced to one instruction on master.

Could you clarify this one?

**Comment by nicolas-grekas** [comment]
from `\PHP_VERSION_ID >= 50400 && \PHP_SESSION_ACTIVE === session_status())`
to `\PHP_SESSION_ACTIVE === session_status())`

**Comment by sroze** [comment]
Hehe, true 👍

**Comment by nicolas-grekas** [comment]
Thank you @sroze.

**Pull request merged**